### PR TITLE
Build Origin with Service Catalog for Install Origin Builds

### LINKED ISSUE #1410: install origin failing consistently for certain jobs
See jobs 141, 142, and 143 for https://ci.openshift.redhat.com/jenkins/view/All/job/test_pull_request_jenkins_sync_plugin/

This job last passed install origins on May 22

See https://ci.openshift.redhat.com/jenkins/view/All/job/test_branch_origin_extended_image_ecosystem/492/

See https://ci.openshift.redhat.com/jenkins/view/All/job/test_branch_origin_extended_image_registry/217/

I see quite a few commits on May 23 and 24 ... is it possible one of those introduced a regression?

@openshift/sig-continuous-delivery @openshift/sig-continuous-infrastructure ptal

### PR DESCRIPTION
**Pull request by adambkaplan**
Builds that install origin need the following:
* build of origin to bring in latest image
* build of the service catalog.

Fixes #1410 until we can take advantage of image streams.

_Note: this will add ~35 minutes to the affected builds_

### CODE REVIEW

File: sjb/config/common/test_cases/origin_installed_release.yml [hunk 1]
```diff
@@ -29,21 +29,6 @@ extensions:
...
-        jobs_repo="/data/src/github.com/openshift/aos-cd-jobs/"
-        git log -1 --pretty=%h > "${jobs_repo}/ORIGIN_COMMIT"
-
-        source hack/lib/init.sh
-        os::build::rpm::get_nvra_vars
-        echo "-${OS_RPM_VERSION}-${OS_RPM_RELEASE}" > "${jobs_repo}/ORIGIN_PKG_VERSION"
-        echo "${OS_GIT_MAJOR}.${OS_GIT_MINOR}" | sed "s/+//" > "${jobs_repo}/ORIGIN_RELEASE"
-        echo "${OS_RPM_VERSION}" | cut -d'.' -f2 > "${jobs_repo}/ORIGIN_PKG_MINOR_VERSION"
-        tag="$( echo "v${OS_GIT_MAJOR}.${OS_GIT_MINOR}" | sed "s/+//" )"
-        echo "${tag}" > "${jobs_repo}/ORIGIN_TAG"
```
Comment (adambkaplan) [inline sjb/config/common/test_cases/origin_installed_release.yml]: removed - duplicates script in [common/test_cases/origin_release.yml](https://github.com/openshift/aos-cd-jobs/blob/master/sjb/config/common/test_cases/origin_release.yml)

File: sjb/config/common/test_cases/origin_release_with_service_catalog.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      script: |-
+        catalog_repo="/data/src/github.com/openshift/service-catalog/"
+        git log -1 --pretty=%h >> "${catalog_repo}/ORIGIN_COMMIT"
+    - type: "script"
+      title: "build the service catalog container image"
+      repository: "service-catalog"
+      timeout: 3600
+      script: |-
+        make build-images
+        docker tag openshift/origin-service-catalog:latest "openshift/origin-service-catalog:$( cat ./ORIGIN_COMMIT )"
```
Comment (adambkaplan) [inline sjb/config/common/test_cases/origin_release_with_service_catalog.yml]: Install origin playbook relies on the service catalog, which is no longer in origin. Sync in the new service catalog and rebuild the image.

File: sjb/config/common/test_cases/origin_installed_release.yml (hunk 1 above)
Comment (gabemontero) [reply sjb/config/common/test_cases/origin_installed_release.yml]: cool

File: sjb/config/common/test_cases/origin_release_with_service_catalog.yml (hunk 2 above)
Comment (gabemontero) [reply sjb/config/common/test_cases/origin_release_with_service_catalog.yml]: yep lines up with what was discussed

Comment (bparees) [reply sjb/config/common/test_cases/origin_release_with_service_catalog.yml]: so there's no existing job we can extend from that already builds/tags the service catalog appropriately?  how are the other e2e jobs running?

**Comment by adambkaplan** [comment]
@bparees updated to use the full ecosystem

**Comment by bparees** [comment]
jenkins updates pushed.  kicked off new runs of the extended jobs.

**Comment by adambkaplan** [comment]
[(3) Extended Tests Monitoring + Online Support + Tech Debt](https://trello.com/c/LokvRxJt/1566-3-extended-tests-monitoring-online-support-tech-debt)

**Pull request merged**